grafana-plugin/integration-tests/globalSetup.ts

```diff
@@ -66,61 +86,55 @@ const configureOnCallPlugin = async (page: Page): Promise<void> => {
  * Borrowed from our friends on the Incident team
  * https://github.com/grafana/incident/blob/main/plugin/e2e/global-setup.ts
  */
-const globalSetup = async (config: FullConfig): Promise<void> => {
-  const { headless } = config.projects[0]!.use;
-  const browser = await chromium.launch({ headless, slowMo: headless ? 0 : 100 });
-  const browserContext = await browser.newContext();
+setup('Configure Grafana OnCall plugin', async ({ request }, { config }) => {
+  /**
+   * Unconditionally marks the setup as "slow", giving it triple the default timeout.
+   * This is mostly useful for the rare case for Cloud Grafana instances where the instance may be down/unavailable
+   * and we need to poll it until it is available
+   */
+  setup.slow();
 
   if (IS_CLOUD) {
-    /**
-     * check that the grafana instance is available. If HTTP 503 is returned it means the
-     * instance is currently unavailable. Poll until it is available
-     */
-    await pollGrafanaInstanceUntilItIsHealthy(browserContext);
+    await grafanaApiClient.pollInstanceUntilItIsHealthy(request);
   }
 
-  const res = await makeGrafanaLoginRequest(browserContext);
-
-  expect(res.ok()).toBeTruthy();
-  await browserContext.storageState({ path: STORAGE_STATE });
-
-  // make sure the plugin has been configured
-  const page = await browserContext.newPage();
+  const adminBrowserContext = await generateLoginStorageStateAndOptionallCreateUser(
+    config,
+    GRAFANA_ADMIN_USERNAME,
+    GRAFANA_ADMIN_PASSWORD,
+    ADMIN_USER_STORAGE_STATE
+  );
+  const adminPage = await adminBrowserContext.newPage();
+  const { request: adminAuthedRequest } = adminBrowserContext;
+
+  await generateLoginStorageStateAndOptionallCreateUser(
```

Comment:
So these calls basically open a new window for each call and perform the login, and in tests you just get a reference on that rolePage?

Reply:
correct 👍 on setup, it will basically generate three files:
- `integration-tests/.auth/admin.json`
- `integration-tests/.auth/editor.json`
- `integration-tests/.auth/viewer.json`

each of the fixtures then create a "browser context" object that is authenticated using one of these three files, depending on which role you select